tests/integration/introduction/test_introduction.py

```diff
@@ -39,3 +40,58 @@ def test_intro_description_not_displayed(self):
 
         # Then description should not be displayed
         self.assertNotIn('qa-intro-description', content)
+
+    def test_ref_start_and_end_date(self):
+
+        # Given survey with ref_p_start_date and ref_p_end_date
+        payload_vars = self._payload_vars_without_ref_end_date()
+        payload_vars['ref_p_end_date'] = '2016-05-31'
+        resp = self._start_survey_with_payload_vars(payload_vars)
+
+        # When on the introduction page
+        content = resp.get_data(True)
+
+        # Then both dates are shown separated by a 'to'
+        self.assertIn('1 May 2016 to 31 May 2016', content)
+
+    def test_ref_start_date_and_missing_ref_end_date(self):
+
+        # Given survey with ref_p_start_date but without ref_p_end_date
+        payload_vars = self._payload_vars_without_ref_end_date()
+        resp = self._start_survey_with_payload_vars(payload_vars)
+
+        # When on the introduction page
+        content = resp.get_data(True)
+
+        # Then one date is shown and there is no 'to'
+        self.assertIn('1 May 2016', content)
+        self.assertNotIn('1 May 2016 to', content)
+
+    @staticmethod
+    def _payload_vars_without_ref_end_date():
```

Comment:
A note going forward I have made create_token easier to be able to these kinds of things. However, it's not in master https://github.com/ONSdigital/eq-survey-runner/pull/1009/files#diff-d7db136e32b2be1e61f07882bcdf8031R25